### PR TITLE
Fix non runway specific STARs not showing

### LINKED ISSUE #1020: [BUG] STAR's won't show up on every airport
**Mod Version**

Release version 0.3.1

**Describe the bug**

STAR's won't show up on every airport.

**To Reproduce**

1. I set up the airplane, ready to file the flightplan in the MCDU.
2. I choose ENGM to EGLL in the Init page
3. When i choose runway 27R at EGLL there where no STAR's available for that runway and the others. I also tried EDDB, but there where no available STAR's there either.

**Expected behavior**

I expect to get the STAR's which is available in the decided airac database.

**Actual behavior**

There where no available STAR's at any airport.

**References**

**Additional context**

Was this working before/when did the issue start occurring?
Im not quite sure when it occured, but i know it worked fine for release 0.1

Is this a problem in the vanilla unmodded game?
Yes, the same problem is in the vanilla version, it looks like it occured when the lates patch was released.

Discord username (if different from GitHub): Amunlar#9633

### PR DESCRIPTION
**Pull request by tyler58546**
Fixes #1020 

**Summary of Changes**

* Fixed STARs not showing on arrivals page when they are not runway specific
* Added workaround for duplicate STARs showing on arrivals page

**Screenshots (if necessary)**

Before:
image

After:
image

**References**

N/A
**Additional context**

N/A

### CODE REVIEW

**Comment by lousybyte** [comment]
This just removes half the STARs from every airport, tested with a small airport like LYNI. Also I didn't see any duplicates on EGLL, from what I gathered the issue is that `arrival.runwayTransitions` is empty for EGLL arrivals.

**Comment by tyler58546** [comment]
> This just removes half the STARs from every airport, tested with a small airport like LYNI. Also I didn't see any duplicates on EGLL, from what I gathered the issue is that `arrival.runwayTransitions` is empty for EGLL arrivals. 
> 
> 

hmm... maybe the duplicates only show up for US airports?

**Comment by lousybyte** [comment]
Here `arrival.runwayTransitions.length` is 0 so that loop is never executed,
https://github.com/flybywiresim/a32nx/blob/03ed84ea3d2d96fa4d3e40b869a8035b7f1822c3/A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_AvailableArrivalsPage.js#L53-L58

thus `matchingArrivals` will be empty here.

https://github.com/flybywiresim/a32nx/blob/03ed84ea3d2d96fa4d3e40b869a8035b7f1822c3/A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_AvailableArrivalsPage.js#L83-L104

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -13,4 +13,5 @@
 6. [EXTERIOR] Fixed issue in engine rotation animation that made it rotate too quickly - @lukecologne (lukecologne)
 7. [PFD] Remove code related to unrelevant aircraft - @1Revenger1 (Avery Black)
 8. [ECAM] Lower ECAM DOOR Page Colour Fix - @nathaninnes (Nathan Innes)
-9. [ND] Add DME distances, VOR/ADF needles and functioning ADF2 - @blitzcaster (bltzcstr)
\ No newline at end of file
+9. [ND] Add DME distances, VOR/ADF needles and functioning ADF2 - @blitzcaster (bltzcstr)
+1. [CDU] Fixed certain STARs not showing on arrivals page - @tyler58546 (tyler58546)
```
Comment (NathanInnes) [inline .github/CHANGELOG.md:17]: Change to current month

**Comment by NathanInnes** [comment]
Haven't had the chance to test yet but is this an issue with the page?
image

**Comment by AdenFlorian** [comment]
@NathanInnes not related to this pr

**Comment by AdenFlorian** [comment]
I removed the /2 fix, since it breaks other airports, and refactored the missing STARs fix to make its intent more clear. Also talked with tyler about it. Tested the missing stars and I can see JULLE5 again!

**Pull request merged**